pkgs/racket-doc/scribblings/reference/hashes.scrbl

```diff
@@ -273,6 +273,75 @@ result:
 
 @see-also-caveats[]}
 
+@defproc[(hash-ref-key [hash hash?]
+                       [key any/c]
+                       [failure-result (failure-result/c any/c)
+                                       (lambda ()
+                                         (raise (make-exn:fail:contract ....)))])
+         any]{
+
+Returns the key held by @racket[hash] that is equivalent to @racket[key]
+according to @racket[hash]'s key-comparison function. If no key is found,
+then @racket[failure-result] is used as in @racket[hash-ref] to determine
+the result.
+
+If @racket[hash] is not an @tech{impersonator}, then the returned key,
+assuming it is found, will be @racket[eq?]-equivalent to the one
+actually retained by @racket[hash]:
+
+@examples[
+#:eval the-eval
+(define original-key "hello")
+(define key-copy (string-copy original-key))
+
+(equal? original-key key-copy)
+(eq? original-key key-copy)
+
+(define table (make-hash))
+(hash-set! table original-key 'value)
+
+(eq? (hash-ref-key table "hello") original-key)
+(eq? (hash-ref-key table "hello") key-copy)
+]
+
+If a mutable hash is updated multiple times using keys that are
+not @racket[eq?]-equivalent but are equivalent according to the
+hash's key-comparison procedure, the hash retains the first one:
+
+@examples[
+#:eval the-eval
+(define original-key "hello")
+(define key-copy (string-copy original-key))
+
+(define table (make-hash))
+(hash-set! table original-key 'one)
+(hash-set! table key-copy 'two)
+
+(eq? (hash-ref-key table "hello") original-key)
+(eq? (hash-ref-key table "hello") key-copy)
+]
+
+Conversely, an immutable hash retains the key that was most-recently
```

Comment:
This behavior could be easily changed to match that of immutable hashes, if we want to do that.